projects/eGolf/Canvas.java

```diff
@@ -1,59 +1,154 @@
+    Timer timer = new Timer(1000 / frameRate, (ActionEvent e) -> {
+      perFrame();
+    });
+    timer.start();
   }
-
+  
   @Override
-    protected void paintComponent(Graphics g) {
-      super.paintComponent(g);
-      var d = getSize();
-      
-      g.setColor(Color.DARK_GRAY);
-      g.fillRect(0, 0, d.width, d.height);
-      g.setColor(Color.RED);
-      g.drawOval(center.x-radius/2, center.y-radius/2-24, radius, radius);
-  }
+  protected void paintComponent(Graphics g) {
+    super.paintComponent(g);
+    var d = getSize();
+    Point mouseLoc = MouseInfo.getPointerInfo().getLocation();
 
-    @Override
-    public void mouseClicked(MouseEvent e) {
-      center.x = e.getX();
-      center.y = e.getY();
-      repaint();
+    // background
+    g.setColor(Color.DARK_GRAY);
+    g.fillRect(0, 0, d.width, d.height);
+    // center
+    g.setColor(Color.white);
+    g.drawOval((int) center.x - 3, (int) center.y - 3, 6, 6);
+    // outer edge
+    g.setColor(Color.RED);
+    g.drawOval((int) center.x - radius, (int) center.y - radius, radius * 2, radius * 2);
+    // drag vect
+    g.setColor(Color.green);
+    if (draggingVect) {
+      g.drawLine((int) center.x, (int) center.y,
+          (int) (center.x + 2 * (center.x - mouseLoc.x)), // stole from stack overflow
+          (int) (center.y + 2 * (center.y - mouseLoc.y)) - 30);
     }
 
-    @Override
-    public void mousePressed(MouseEvent e) {
-      System.out.println("pressed");
+    // motion in 1s
+    g.setColor(Color.blue);
+    g.drawLine((int) center.x, (int) center.y, (int) (center.x + vect.x), (int) (center.y + vect.y));
+    // new wheel
+
+    if (draggingWheel) {
+      int r = twoDist(mouseLoc, wheelStart);
+      Point tl = topLeft(wheelStart, radius);
+      g.setColor(Color.yellow);
+      g.drawOval(tl.x, tl.y, r, r);
     }
+    // drawing wheels
 
-    @Override
-    public void mouseReleased(MouseEvent e) {
-      System.out.println("releaased");
+    for (Map.Entry<Point, Integer> entry : wheels.entrySet()) {
+      Point center = entry.getKey();
+      int rad = entry.getValue();
+      g.setColor(Color.orange);
+      g.drawOval(center.x - rad, center.y - rad, rad*2, rad*2);
     }
 
-    @Override
-    public void mouseEntered(MouseEvent e) {
-      System.out.println("clicked on");
+  }
+
+  private int twoDist (Point a, Point b) {
+    return (int) Math.hypot(a.x-b.x, a.y-b.y);
+  }
+
+  private Point topLeft (Point center, int rad) {
+    return new Point(center.x - rad, center.y-rad-30);
+  }
+
+  private void wallColls(Point dim) {
+    if (center.x + radius > dim.x) {
+      center.x = dim.x - radius;
+      vect.x *= -1;
+    }
+    if (center.x - radius < 0) {
+      center.x = radius;
+      vect.x *= -1;
+    }
+    if (center.y + radius > dim.y) {
+      center.y = dim.y - radius;
+      vect.y *= -1;
     }
+    if (center.y - radius < 0) {
+      center.y = radius;
+      vect.y *= -1;
+    }
+  }
+
+  private void perFrame() {
```

Comment:
for the whole project it would be helpful to have comments that would let the reader know what each thing was for.